akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -518,19 +606,94 @@ final class ClusterReceptionistSettings(
   def withResponseTunnelReceiveTimeout(responseTunnelReceiveTimeout: FiniteDuration): ClusterReceptionistSettings =
     copy(responseTunnelReceiveTimeout = responseTunnelReceiveTimeout)
 
+  def withHeartbeat(
+    heartbeatInterval: FiniteDuration,
+    acceptableHeartbeatPause: FiniteDuration,
+    failureDetectionInterval: FiniteDuration): ClusterReceptionistSettings =
+    copy(
+      heartbeatInterval = heartbeatInterval,
+      acceptableHeartbeatPause = acceptableHeartbeatPause,
+      failureDetectionInterval = failureDetectionInterval)
+
   private def copy(
     role: Option[String] = role,
     numberOfContacts: Int = numberOfContacts,
-    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout): ClusterReceptionistSettings =
-    new ClusterReceptionistSettings(role, numberOfContacts, responseTunnelReceiveTimeout)
-
+    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout,
+    heartbeatInterval: FiniteDuration = heartbeatInterval,
+    acceptableHeartbeatPause: FiniteDuration = acceptableHeartbeatPause,
+    failureDetectionInterval: FiniteDuration = failureDetectionInterval): ClusterReceptionistSettings =
+    new ClusterReceptionistSettings(
+      role,
+      numberOfContacts,
+      responseTunnelReceiveTimeout,
+      heartbeatInterval,
+      acceptableHeartbeatPause,
+      failureDetectionInterval)
 }
 
 /**
  * Marker trait for remote messages with special serializer.
  */
 sealed trait ClusterClientMessage extends Serializable
 
+/**
+ * Declares a super type for all events emitted by the `ClusterReceptionist`.
+ * in relation to cluster clients being interacted with.
+ * Events are published to the Akka event stream (system.eventStream). Given
+ * that these messages are transmitted across the event bus there is no need
+ * for serialization.
+ */
+sealed trait ClusterClientInteraction extends NoSerializationVerificationNeeded {
+  val clusterClient: ActorRef
+  val clusterReceptionist: ActorRef
+}
```

Comment:
I’m now just marking the specific classes.

> On 9 May 2016, at 4:43 PM, Christopher Hunt <huntchr@gmail.com> wrote:
> 
> My goal here is to state that the messages are not intended for transmission beyond where the event bus can be accessed… and the event bus can only be accessed locally. Is that bad?
> 
>> On 9 May 2016, at 4:03 PM, Patrik Nordwall <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> In akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala <https://github.com/akka/akka/pull/20462#discussion_r62452938>:
>> 
>> >  }
>> >  
>> >  /**
>> >   * Marker trait for remote messages with special serializer.
>> >   */
>> >  sealed trait ClusterClientMessage extends Serializable
>> >  
>> > +/**
>> > + * Declares a super type for all events emitted by the `ClusterReceptionist`.
>> > + * in relation to cluster clients being interacted with.
>> > + * Events are published to the Akka event stream (system.eventStream). Given
>> > + * that these messages are transmitted across the event bus there is no need
>> > + * for serialization.
>> > + */
>> > +sealed trait ClusterClientInteraction extends NoSerializationVerificationNeeded {
>> > +  val clusterClient: ActorRef
>> > +  val clusterReceptionist: ActorRef
>> > +}
>> But if someone wants to write a serializer for them it's not wrong to send them over remoting. Please don't sprinkle NoSerializationVerificationNeeded to everything, it should only be used when really needed for testing purposes.
>> 
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub <https://github.com/akka/akka/pull/20462/files/f6056876db54608dc4b055f3af47295a87c594bf#r62452938>
>